From 71b267966dfc74ae064363ec971e435c90b31209 Mon Sep 17 00:00:00 2001
From: Khoa Hoang <admin@khoahoang.com>
Date: Tue, 23 Nov 2021 01:33:58 -0800
Subject: [PATCH] lib: hardware-mapping.c: add cubie-hat GPIO mapping

---
 lib/hardware-mapping.c | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/lib/hardware-mapping.c b/lib/hardware-mapping.c
index 758b428..7fc4810 100644
--- a/lib/hardware-mapping.c
+++ b/lib/hardware-mapping.c
@@ -213,6 +213,41 @@ struct HardwareMapping matrix_hardware_mappings[] = {
     .p0_b2         = GPIO_BIT(25),
   },
 
+  /*
+   * LED cube modified hat
+   */
+  {
+    .name          = "cubie-hat",
+
+    .output_enable = GPIO_BIT(12), /* swap 18 -> 12 from regular */
+    .clock         = GPIO_BIT(17),
+    .strobe        = GPIO_BIT(4),
+
+    /* Address lines */
+    .a             = GPIO_BIT(22),
+    .b             = GPIO_BIT(23),
+    .c             = GPIO_BIT(24),
+    .d             = GPIO_BIT(25),
+    .e             = GPIO_BIT(15),  /* RxD kept free unless 1:64 */
+
+    /* Chain 0 */
+    .p0_r1         = GPIO_BIT(14), /* masks: SPI0_CE1 swap 12 -> 14 from regular */
+    .p0_g1         = GPIO_BIT(5),
+    .p0_b1         = GPIO_BIT(6),
+    .p0_r2         = GPIO_BIT(16), /* swap 19 -> 16 from regular */
+    .p0_g2         = GPIO_BIT(13),
+    .p0_b2         = GPIO_BIT(26), /* swap 20 -> 26 from regular */
+
+    /* Chain 1 */
+    .p1_r1         = GPIO_BIT(11),  /* masks: SPI0_SCKL  */
+    .p1_g1         = GPIO_BIT(27),  /* Not on RPi1, Rev1; use "regular-pi1" instead */
+    .p1_b1         = GPIO_BIT(7),   /* masks: SPI0_CE1   */
+    .p1_r2         = GPIO_BIT(8),   /* masks: SPI0_CE0   */
+    .p1_g2         = GPIO_BIT(9),   /* masks: SPI0_MISO  */
+    .p1_b2         = GPIO_BIT(10),  /* masks: SPI0_MOSI  */
+  },
+
+
 #ifdef ENABLE_WIDE_GPIO_COMPUTE_MODULE
   /*
    * Custom pin-out for compute-module
-- 
2.33.1

